flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/DefaultDispatcherResourceManagerComponentFactory.java

```diff
@@ -69,7 +69,7 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
```

Comment:
I'm not sure about replacing `Executor` with `ExecutorService` for all these places.

Looking into `Fabric8FlinkKubeClient`, I think the only reason `FlinkKubeClient` uses `ExecutorService` rather than `Executor` is that, it will shutdown and release the thread pool in `close`. That is because the client uses a dedicated thread pool, and if it does not shutdown the thread pool, other components won't do that either.

Since now the client uses a shared IO executor, it should no longer shutdown the executor on close. Otherwise, all the other components using that IO executor would be affected. That means we can replace `ExecutorService` with `Executor` for `FlinkKubeClient`.